logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogStreamImpl.java

```diff
@@ -382,4 +403,11 @@ public void onUnrecoverableFailure(final HealthReport report) {
 
     T create(int partitionId, Dispatcher dispatcher);
   }
+
+  private static final class ShouldCloseAppenderException extends RuntimeException {
```

Comment:
:wrench: Can we rename this to something like `LogStorageAppenderClosed`? When this exception appears, the appender is already closed so I think "shouldClose" is confusing. Similarly, I think the exception message could be something along the lines of "LogStorageAppender was closed before opening succeeded", the important part being that a close wasn't just requested but has already happened.

Reply:
Done